src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseExportService.java

```diff
@@ -156,19 +168,54 @@ public Path exportProgrammingExerciseInstructorMaterial(ProgrammingExercise exer
     }
 
     /**
-     * Export instructor repositories and optionally students' repositories in a zip file.
+     * In case the problem statement contains embedded files, they need to be part of the zip, so they can be imported again.
      *
+     * @param exercise        the programming exercise that is exported
+     * @param outputDir       the directory where the content of the export is stored
+     * @param pathsToBeZipped the paths that should be included in the zip file
+     */
+
+    private void copyEmbeddedFiles(ProgrammingExercise exercise, Path outputDir, List<Path> pathsToBeZipped) throws IOException {
+        List<String> embeddedFiles = new ArrayList<>();
+
+        Matcher matcher = Pattern.compile(EMBEDDED_FILE_REGEX).matcher(exercise.getProblemStatement());
+        while (matcher.find()) {
+            embeddedFiles.add(matcher.group());
+        }
+        log.debug("Found embedded files:{} ", embeddedFiles);
+        Path embeddedFilesDir = null;
+        if (!embeddedFiles.isEmpty()) {
+            embeddedFilesDir = outputDir.resolve("files");
+            if (!Files.exists(embeddedFilesDir)) {
+                Files.createDirectory(embeddedFilesDir);
+            }
+            pathsToBeZipped.add(embeddedFilesDir);
+        }
+        for (String embeddedFile : embeddedFiles) {
+            String filePath = embeddedFile.substring(embeddedFile.indexOf("(") + 1, embeddedFile.indexOf(")"));
+            String fileName = filePath.replace(API_MARKDOWN_FILE_PATH, "");
+            Path imageFilePath = Path.of(FilePathService.getMarkdownFilePath(), fileName);
+            Path imageExportPath = embeddedFilesDir.resolve(fileName);
+            Files.copy(imageFilePath, imageExportPath);
+        }
+
+    }
+
+    /**
+     * Export instructor repositories and optionally students' repositories in a zip file.
+     * <p>
      * The outputDir is used to store the zip file and temporary files used for zipping so make
      * sure to delete it if it's no longer used.
      *
      * @param exercise              the programming exercise
      * @param includingStudentRepos flag for including the students repos as well
      * @param outputDir             the path to a directory that will be used to store the zipped programming exercise.
+     * @param workingDir            the path to a directory that will be used to store the cloned repositories.
      * @param exportErrors          List of failures that occurred during the export
      * @param reportData            List of all exercises and their statistics
      * @return the path to the zip file
      */
-    public Path exportProgrammingExerciseRepositories(ProgrammingExercise exercise, Boolean includingStudentRepos, Path outputDir, List<String> exportErrors,
+    public Path exportProgrammingExerciseRepositories(ProgrammingExercise exercise, Boolean includingStudentRepos, Path workingDir, Path outputDir, List<String> exportErrors,
```

Comment:
Why can't we use the output directory for working as well?

Reply:
because we zip all the content in the directory,  If we do not distinguish the directory the cloned repos are part of the final zip file

Reply:
But couldn't we simply use two subfolders in the directory that is provided to us?
This would keep the parameter list shorter

Reply:
no, because the content of the zip is all content recursively, so even if it's a subdirectory it's included.